test/changed.sh

```diff
@@ -75,6 +75,12 @@ for directory in ${CHANGED_FOLDERS}; do
     kubectl get deployments --namespace ${NAMESPACE}
     kubectl get endpoints --namespace ${NAMESPACE}
     helm test ${RELEASE_NAME}
+
+    # If tests fail capture output to aide in debugging the test
+    if [ $? -ne 0 ]; then
+      kubectl logs ${RELEASE_NAME} --namespace ${NAMESPACE}
```

Comment:
This doesn't seem right. It's actually asking you to get the logs from the Pod that was created not necessarily the release name. 

For example when running the Jenkins test I get:
```
RUNNING: j-ui-test-icokn
FAILED: j-ui-test-icokn, run `kubectl logs j-ui-test-icokn --namespace default` for more info
Error: 1 test(s) failed
``` 

Where the release was named `j`

I don't think we have enough convention across the tests to say that all the test pods will be called the same as the release name.

In the Jenkins chart it uses the following naming convention:
https://github.com/kubernetes/charts/blob/master/stable/jenkins/templates/jenkins-test.yaml#L4

Not sure how we can proceed unless we enforce that all test pods are called $RELEASE-test or something like that.

Reply:
What we may need is for kubernetes/helm#1534 to land for the status command. Then we can iterate over the pods and get the logs. Thoughts?

Reply:
That sounds like a good approach.

Reply:
I just checked and I don't think kubernetes/helm#1534 would solve the issue. `helm status` won't show the test pod...

We should be able to do this with a fun go template:
```
TEST_POD_NAMES=$(kubectl get po --show-all -o go-template='{{range .items}}{{ $hook := print (index .metadata.annotations "helm.sh/hook") }}{{ if or (eq $hook "test-success") (eq $hook "test-failure") }}{{printf "%s\n" .metadata.name}}{{end}}{{end}}')
```

...and then loop through those names and print logs. We could even extend the go template there to check for pod phase, but this would need to be "Failed" in the case of test-success and "Succeeded" in the case of test-failure. Probably easier to just print all.

Reply:
Shouldn't we get logs from the test pod and all the other pods running for the release as well?

Working on this now so it's relevant.